packages/vc-verification/src/issuer-verification.ts

```diff
@@ -0,0 +1,255 @@
+// @ts-ignore
+import * as jwt from 'jsonwebtoken';
+import { utils, providers } from 'ethers';
+import { ProofVerifier } from '@ew-did-registry/claims';
+import { Resolver } from '@ew-did-registry/did-ethr-resolver';
+import { RegistrySettings } from '@ew-did-registry/did-resolver-interface';
+import { RoleDefinitionResolverV2__factory } from '@energyweb/credential-governance/ethers/factories/RoleDefinitionResolverV2__factory';
+import { RoleDefinitionResolverV2 } from '@energyweb/credential-governance/ethers/RoleDefinitionResolverV2';
+import { StorageSettings } from './models';
+import { CredentialResolver } from '.';
+import { IVerifiableCredential, VerificationResult } from './models';
+
+export class IssuanceVerification {
+  private _provider: providers.Provider;
+  private _resolver: Resolver;
+  private _registrySetting: RegistrySettings;
+  private _roleDefResolver: RoleDefinitionResolverV2;
+  private _credentialResolver: CredentialResolver;
+
+  /**
+   *
+   * @param provider
+   * @param roleDefResolverAddr
+   * @param registrySetting
+   * @param storageSettings
+   */
+  constructor({
+    provider,
+    roleDefResolverAddr,
+    registrySetting,
+    storageSettings,
+  }: {
+    provider: providers.Provider;
+    roleDefResolverAddr: string;
+    registrySetting: RegistrySettings;
+    storageSettings: StorageSettings;
+  }) {
+    this._provider = provider;
+    this._registrySetting = registrySetting;
+    this._roleDefResolver = RoleDefinitionResolverV2__factory.connect(
+      roleDefResolverAddr,
+      this._provider
+    );
+    this._resolver = new Resolver(this._provider, this._registrySetting);
+    this._credentialResolver = new CredentialResolver(storageSettings);
```

Comment:
Instead of hardcoding the `CredentialResolver` construction, I think it would be better to accept the `CredentialResolver` object directly. Or even better would be to accept an `ICredentialResolver` interface so that we could swap out the implementation in the future. But if that's overkill for now, no worries.

Reply:
I guess the `CredentialResolver` object seems to be reasonable as one could have flexibility to decide the approach / configuration for resolution of credential, it could be either from IPFS or some hosted / published accreditation based on governance. So it would be up to the caller or the need of the resolution to decide on the way of initialisation of `CredentialResolver` and changes could be made directly to `CredentialResolver` as per need, while `verification` code remains intact.